### PR TITLE
Handle already-initialized components

### PR DESCRIPTION
**Pull request by EtienneBruines**
Sometimes the components within the configuration may already be configured. 

This change adds support for following config syntax: 

```php
return [
    'components' => [
        'client' => new \GuzzleHttp\Client(),
    ],
]
```

### CODE REVIEW

File: src/ServiceMap.php [hunk 1]
```diff
@@ -43,6 +43,11 @@ public function __construct(string $configPath)
         }
 
         foreach ($config['components'] ?? [] as $id => $component) {
+            if (!is_array($component)) {
```
Comment (akondas) [inline src/ServiceMap.php]: :thinking: maybe also `is_object`?

File: src/ServiceMap.php [hunk 2]
```diff
@@ -43,6 +43,11 @@ public function __construct(string $configPath)
...
+                $this->components[$id] = $component;
+                continue;
```
Comment (akondas) [inline src/ServiceMap.php]: what about test for this case?

File: src/ServiceMap.php (hunk 1 above)
Comment (EtienneBruines) [reply src/ServiceMap.php]: I'm guessing `is_object` implies `!is_array` and that `is_object` is actually more reliable. 

I will update this and add a test-case for the edge-case where `!is_array`-behavior is not desired. :smile:

**Comment by EtienneBruines** [comment]
Hello @akondas, thank you for your code review. It is much appreciated. 

As per your suggestion, I changed `!is_array` to `is_object`. Additionally, I added a check to verify that if it's not an object, it should at least be an array - otherwise the code will crash in a weird manner anyways. Now a more useful error message is thrown.

Two tests were added: 
- `testThrowExceptionWhenComponentHasInvalidValue` to verify the exception is thrown when the component value was invalid (e.g. a numeric value)
- `testItLoadsServicesAndComponents` (updated) to verify that the additional component `customInitializedComponent` is recognized.

I wasn't quite sure what the `getComponentIdentityClassById` method was supposed to do, so I didn't change it. I had to add a special condition for `getComponentClassById` to make it perform as expected. 

If you have any points of improvements, don't hesitate to mention them :smile:

**Comment by akondas** [comment]
Great, thanks for tests :+1:
